data-forwarder/src/main.rs

```diff
@@ -123,11 +124,15 @@ fn main() -> Result<(), std::boxed::Box<dyn std::error::Error>> {
     let curr_dir = env::current_dir()?;
     let repo = Repository::discover(curr_dir)?;
     let head = repo.head()?;
-    let git_branch_name = if head.is_branch() {
-        head.shorthand().map(|t| t.to_string())
-    } else {
-        None
-    };
+    let git_branch_name_cmd = Command::new("git")
+        .arg("name-rev")
+        .arg("--name-only")
+        .arg("--exclude=*HEAD*")
+        .arg("HEAD")
+        .output()
+        .expect("failed to execute process");
```

Comment:
If we for some reason can't execute git, we still want to collect the rest of the scan information. If we replace expect with `.ok()`, it'll convert the Result<T, E> to an Option<T>, with a value of None if there was an error. Keep the same behaviour as we currently have, but with a better effort at detecting the current branch.

```suggestion
        .ok();
```

Reply:
You could then continue this chain by calling map to convert the contents of the stdout field to a String if the command was successful and still pass the option to the ToolReport struct initialiser